pgroll-integration.mdx

```diff
@@ -0,0 +1,102 @@
+
+The only things that change for these new Postgres enabled databases in terms of our API are primary key names and internal meta data field names. The `id` field has been renamed to `xata_id`, and `xata.createdAt`, `xata.updatedAt`, and `xata.version` fields have been renamed to `xata_createdat`, `xata_updatedat`, and `xata_version`.
+
+![New internal column names](/images/pgroll-integration/rename.png)
+
+## Technical challenges
+
+Integrating pgroll within our web application and developer packages has posed some interesting technical challenges. Some of the main challenges we encountered were:
+
+- **Backwards compatibility with existing migration workflows**
+  - We wanted to continue to support databases that did not have direct access to Postgres enabled so as not to break existing users’ queries. In order to maintain backwards compatibility for existing databases, we had to figure out some way to communicate to the backend whether we wanted to use the new APIs or the old ones. We settled on a combination of feature flags and request headers.
+- **Support for arbitrary Postgres-native column types**
+  - Prior to the integration, there was a relatively small list of column types that we knew a column could be. Arbitrary SQL migrations over pgroll greatly expanded that list as users could create their own Postgres native column types. We had to do a lot of mapping between these new native Postgres types and the known Xata column types, as our web application has custom React components for displaying, editing, and adding data according to a column’s Xata type. Some of these new Postgres native types could not be mapped into our existing Xata types as they do not work with our Data API, so figuring that out and handling those cases gracefully was a challenge.
+- **Detecting when a migration is complete**
+  - pgroll does not allow for referencing items created in previous operations in subsequent operations within the same migration. This means that running an `alter_column` operation after an `add_column` operation, or running a `rename_constraint` operation (requires a table name) after a `rename_table` operation within the same migration request is not currently possible. This is due to the fact that the first operation in the list is guaranteed to compete before the second operation completes, but the first operation is not guaranteed to complete before the second operation starts. Since the backend provides an asynchronous `apply` endpoint which responds with a `jobId`, we decided that the frontend would use the returned `jobId` as input to a second new endpoint called `status` which would return the state of that migration. The frontend polls the status endpoint until the migration in question succeedes or fails before moving forward with dependent migrations.
+
+## Path to General Availability
```

Comment:
@tsg @exekias @andrew-farries @eemmiillyy keep me honest with GA plans. Should I remove or add anything?

Reply:
LGTM! added a small comment